.github/workflows/docs.yml

```diff
@@ -0,0 +1,69 @@
+name: docs
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'mkdocs' # Remove after merge
+      - 'v*'
+    tags:
+      - 'v*'
+    paths:
+      - '.github/workflows/docs.yml'
+      - 'docs/**'
+      - 'mkdocs.yml'
+  pull_request:
+    branches:
+      - 'main'
+      - 'v*'
+    paths:
+      - '.github/workflows/docs.yml'
+      - 'docs/**'
+      - 'mkdocs.yml'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      -
+        name: Prepare
+        id: prepare
+        run: |
+          VERSION=edge
+          RELEASE=false
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/v}
+          fi
+          if [[ $VERSION =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+            RELEASE=true
+          fi
+          echo ::set-output name=release::${RELEASE}
+      -
+        name: Build mkdocs Docker image
+        run: |
+          docker build -t mkdocs -f ./docs/Dockerfile ./
+      -
+        name: Build docs
+        run: |
+          docker run --rm -v "$(pwd):/docs" mkdocs build --strict
+          sudo chown -R $(id -u):$(id -g) ./site
+      -
+        name: Check GitHub Pages status
+        uses: crazy-max/ghaction-github-status@v1
+        with:
+          pages_threshold: major_outage
+      -
+        name: Deploy
+        #if: success() && github.event_name != 'pull_request' && (endsWith(github.ref, github.event.repository.default_branch) || steps.prepare.outputs.release == 'true')
+        if: success() && github.event_name != 'pull_request' # Remove after merge and use the previous condition
```

Comment:
Remove this line when merged and uncomment the previous one (was here for testing purpose and demo)